### PR TITLE
project(spm): use exact dependency versions

### PR DESCRIPTION
**Pull request by dskuza**
## Summary

Update usage of Swift Package Manager to utilize exact versions instead of up to next major. This is in hopes that this helps us resolve issues with conflicting `Package.resolved` files, or repeatedly seeing file change when switching branches.

## References 

IN-1126

## Implementation Details

- Update both `Package.swift` files
  - Additionally update all files to use latest APIs and Swift 5.7
- Update package(s) within Pocket project settings

## QA Steps

- [x] Passing bitrise build

## PR Checklist:

- [x] Self Review (review, clean up, documentation, run tests)
- [x] Basic Self QA

### CODE REVIEW

**Review by timc-mozilla — APPROVED** [review]

File: Pocket.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -86,8 +86,8 @@
       "kind" : "remoteSourceControl",
       "location" : "https://github.com/snowplow/snowplow-objc-tracker",
       "state" : {
-        "revision" : "ec912a566073615ebcc9dc3a4c6b96a1cc6ed7e7",
-        "version" : "4.1.0"
+        "revision" : "36b53bb2c1b72bde8c531dc91fa75b2925e77e74",
+        "version" : "4.0.1"
```
Comment (cyndichin) [inline Pocket.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:90]: should this be `4.1.0`?

File: Pocket.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 2]
```diff
@@ -149,8 +149,8 @@
       "kind" : "remoteSourceControl",
       "location" : "https://github.com/SvenTiigi/YouTubePlayerKit.git",
       "state" : {
-        "revision" : "fa70945f48780b2f78102c5905b6a65b2531b618",
-        "version" : "1.1.12"
+        "revision" : "4fe5c72b073af99cee33a5795ce97e03aac2b227",
+        "version" : "1.1.5"
```
Comment (cyndichin) [inline Pocket.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:153]: `1.1.12`

File: Pocket.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 3]
```diff
@@ -72,6 +72,15 @@
...
     },
+    {
+      "identity" : "sdwebimage",
+      "kind" : "remoteSourceControl",
+      "location" : "https://github.com/SDWebImage/SDWebImage.git",
+      "state" : {
+        "revision" : "4178d12a44691182c2eb79c70281b14cb73f4cbf",
+        "version" : "5.14.3"
+      }
+    },
```
Comment (cyndichin) [inline Pocket.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:83]: how come this got added vs a version change?

**Review by cyndichin — APPROVED** [review]

**Review by timc-mozilla — APPROVED** [review]

File: Pocket.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 3 above)
Comment (dskuza) [reply Pocket.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:83]: it must be a dependency of a dependency, i'm just not sure which one (i don't think spm can show a dependency tree)

**Review by cyndichin — APPROVED** [review]

**Pull request merged**